components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/OrientAsyncHelper.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.storage;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nullable;
+
+import org.sonatype.sisu.goodies.common.ComponentSupport;
+
+import com.google.common.base.Throwables;
+import com.orientechnologies.orient.core.command.OCommandResultListener;
+import com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx;
+import com.orientechnologies.orient.core.record.impl.ODocument;
+import com.orientechnologies.orient.core.sql.query.OSQLNonBlockingQuery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Helper class to wrap Orient SQL SELECT queries into {@link Iterable}s.
+ *
+ * @since 3.0
+ */
+class OrientAsyncHelper
+    extends ComponentSupport
+{
+  private static final int BUFFER_SIZE = 128;
+
+  private static final long TIMEOUT_SECONDS = 60L;
```

Comment:
Would be useful if this was configurable as an optional `asyncIterable` parameter for situations where we don't mind waiting longer, can always keep 60s as the default in the simpler form of `asyncIterable`.